src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs

```diff
@@ -63,7 +68,9 @@ public class LanguageServer
         /// </summary>
         public NLHandler NL2FxImplementation { get; set; }
 
-        public LanguageServer(SendToClient sendToClient, IPowerFxScopeFactory scopeFactory, Action<string> logger = null)
+        internal readonly INLHandlerFactory _nlHandlerFactory;
+
+        public LanguageServer(SendToClient sendToClient, IPowerFxScopeFactory scopeFactory, Action<string> logger = null, INLHandlerFactory nlHandlerFactory = null)
```

Comment:
> public LanguageServer(SendToClient sendToClient, IPowerFxScopeFactory scopeFactory, Action<string> logger = null, INLHandlerFactory nlHandlerFactory = null) [](http://example.com/codeflow?start=7&length=157)

Can we add new ctor, to avoid binary breaking change? 
 
or

@MikeStall Can we take this breaking change and remove NL2FxImplementation entirely since its not thread safe?

Reply:
This doesn't seem to introduce any compile time error in Doc server atleast. For example, Action<void> logger = null was not part of language server ctor initially and it was added as an optional param by Luc. It never broke anything.  The new nlhandler factory is following the same and i do not think it would break any consumer of this sdk

Reply:
It's a binary-breaking change.  But not a source-breaking change
DocServer didn't hit it because you recompiled - but hosts that just pull updated nuget w/o recompile will break. 
 
Can we make it a init property?